.github/workflows/codespell.yml

```diff
@@ -0,0 +1,15 @@
+# GitHub Action to automate the identification of common misspellings in text files
+# https://github.com/codespell-project/codespell
+# https://github.com/codespell-project/actions-codespell
+name: codespell
+on: [push, pull_request]
+jobs:
+  codespell:
+    name: Check for spelling errors
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: codespell-project/actions-codespell@master
+        with:
+          skip: ./bin/trace,./hl/tools/h5watch/h5watch.c,./tools/test/h5jam/tellub.c
+          ignore_words_list: isnt,inout,nd,parms,parm,ba,offsetP,ser,ois,had,fiter,fo,clude,refere,minnum,offsetp,creat,ans:,eiter,lastr,minum,ans,isn't,ifset,sur,trun,dne,tthe,hda,filname,te,htmp
```

Comment:
I'm not a fan of running this automatically. There were several instances of the spell checker making mistakes in this PR, some of which were in actual code and not comments.

Reply:
I disagree, the false positives are relatively low. It should be checked as part of the PR, if it becomes a major issue then we can revisit it.

Reply:
FWIW the action doesn't do anything (like fix things) by default aside from fail if it finds typos. I run it on a number of repos and it doesn't often flag new false positives and when it does it's easy enough to add them to the ignore or skip lists as appropriate.